audio/Makefile

```diff
@@ -1,55 +1,23 @@
-PHONY := all package clean
-rwildcard=$(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2) $(filter $(subst *,%,$2),$d))
-
-CC := arm-vita-eabi-gcc
-CXX := arm-vita-eabi-g++
-STRIP := arm-vita-eabi-strip
-
-PROJECT_TITLE := VitaSDK sample
-PROJECT_TITLEID := VSDK00001
-
-PROJECT := vita_sample
-CXXFLAGS += -std=c++11 -I../common
-CFLAGS += -std=gnu11 -I../common
-
-LIBS := -lSceDisplay_stub -lSceCtrl_stub  -lSceAudio_stub 
-
-SRC_C :=$(call rwildcard, src/, *.c)
-SRC_CPP :=$(call rwildcard, src/, *.cpp)
-
-OBJ_DIRS := $(addprefix out/, $(dir $(SRC_C:src/%.c=%.o))) $(addprefix out/, $(dir $(SRC_CPP:src/%.cpp=%.o)))
-OBJS := $(addprefix out/, $(SRC_C:src/%.c=%.o)) $(addprefix out/, $(SRC_CPP:src/%.cpp=%.o))
-
+CFLAGS += -I../common
+STUBS := Display Ctrl Audio
+LIBS := -lm
+#no need to modify the following
 
+PROJECT := $(shell basename $(CURDIR))
+PHONY := all package clean
 all: package
-
 package: $(PROJECT).vpk
-
-$(PROJECT).vpk: eboot.bin param.sfo
-	vita-pack-vpk -s param.sfo -b eboot.bin $(PROJECT).vpk
-	
-eboot.bin: $(PROJECT).velf
-	vita-make-fself $(PROJECT).velf eboot.bin
-
-param.sfo:
-	vita-mksfoex -s TITLE_ID="$(PROJECT_TITLEID)" "$(PROJECT_TITLE)" param.sfo
-
+clean:
+	rm -rf $(PROJECT).*
+
+$(PROJECT).vpk: $(PROJECT).sfo $(PROJECT).fself
+	vita-pack-vpk -s $< -b $(word 2,$^) $(foreach A,$(wildcard sce_sys/*.*) $(wildcard sce_sys/*/*/*.*),--add $A=$A) $@
+$(PROJECT).fself: $(PROJECT).velf
+	vita-make-fself $< $@
+$(PROJECT).sfo:
+	vita-mksfoex -s TITLE_ID="VSDK00001" "$(PROJECT)" $@
 $(PROJECT).velf: $(PROJECT).elf
-	$(STRIP) -g $<
+	arm-vita-eabi-strip -g $<
 	vita-elf-create $< $@
-
-$(PROJECT).elf: $(OBJS)
-	$(CXX) -Wl,-q $(CFLAGS)  -o $@ $^ $(LIBS)
-
-$(OBJ_DIRS):
-	mkdir -p $@
-
-out/%.o : src/%.cpp | $(OBJ_DIRS)
-	arm-vita-eabi-g++ -c $(CXXFLAGS) -o $@ $<
-
-out/%.o : src/%.c | $(OBJ_DIRS)
-	arm-vita-eabi-gcc -c $(CFLAGS) -o $@ $<
-
-clean:
-	rm -f $(PROJECT).velf $(PROJECT).elf $(PROJECT).vpk param.sfo eboot.bin $(OBJS)
-	rm -r $(abspath $(OBJ_DIRS))
+$(PROJECT).elf: $(wildcard src/*.c)
+	arm-vita-eabi-gcc -Wl,-q $(CFLAGS) -o $@ $^ $(LIBS) $(patsubst %,-lSce%_stub,$(STUBS))
```

Comment:
Correct, Is there any other library suffix bessie _stub and _stub_weak ?
A fall back would be to use $(LIBS) which can hold any exotic library name